src/cunumeric/scalar_reduction_buffer.h

```diff
@@ -0,0 +1,60 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#pragma once
+
+#include "core/cuda/cuda_help.h"
+#include "core/data/buffer.h"
+
+namespace cunumeric {
+
+template <typename REDOP>
+class ScalarReductionBuffer {
+ private:
+  using LHS = typename REDOP::LHS;
+  using RHS = typename REDOP::RHS;
+
+ public:
+  ScalarReductionBuffer(cudaStream_t stream) : buffer_(legate::create_buffer<LHS>(1))
```

Comment:
Since this is again only going to run on the device, do we want to explicitly pass GPU_FB_MEM to create_buffer to make it clearer what is happening? Otherwise this is using the default kind = NO_MEMKIND, which seems potentially fragile to rely on get_executing_processor() returning TOC_PROC to allocate this in the right place.

Reply:
https://github.com/nv-legate/cunumeric/pull/509/commits/2533b0d994964355f64c0010e12bdc8df41eab09